component/core/src/test/java/org/exoplatform/social/core/listeners/ActivityAttachmentMetadataListenerTest.java

```diff
@@ -0,0 +1,165 @@
+    identityManager = getContainer().getComponentInstanceOfType(IdentityManager.class);
+    activityManager = getContainer().getComponentInstanceOfType(ActivityManager.class);
+    metadataService = getContainer().getComponentInstanceOfType(MetadataService.class);
+    metadataDAO = getContainer().getComponentInstanceOfType(MetadataDAO.class);
+    if (metadataService.getMetadataTypeByName(METADATA_TYPE_NAME) == null) {
+      MetadataTypePlugin userMetadataTypePlugin = new MetadataTypePlugin(newParam(2658, METADATA_TYPE_NAME)) {
+        @Override
+        public boolean isAllowMultipleItemsPerObject() {
+          return true;
+        }
+
+        @Override
+        public boolean isShareable() {
+          return true;
+        }
+      };
+      metadataService.addMetadataTypePlugin(userMetadataTypePlugin);
+    }
+
+    johnIdentity = identityManager.getOrCreateUserIdentity("john");
+    tearDownActivityList = new ArrayList<>();
+    tearDownSpaceList = new ArrayList<>();
+  }
+
+  @Override
+  public void tearDown() throws Exception {
+    end();
+    begin();
+    identityManager.deleteIdentity(johnIdentity);
+    metadataDAO.deleteAll();
+
+    for (ExoSocialActivity activity : tearDownActivityList) {
+      try {
+        activityManager.deleteActivity(activity.getId());
+      } catch (Exception e) {
+        LOG.warn("can not delete activity with id: " + activity.getId());
+      }
+    }
+
+    for (Space space : tearDownSpaceList) {
+      Identity spaceIdentity = identityManager.getOrCreateIdentity(SpaceIdentityProvider.NAME, space.getPrettyName());
+      if (spaceIdentity != null) {
+        identityManager.deleteIdentity(spaceIdentity);
+      }
+      spaceService.deleteSpace(space);
+    }
+
+    super.tearDown();
+  }
+
+  public void testDeleteActivity() throws ObjectAlreadyExistsException{
+    String activityTitle = "activity title";
+    String userId = johnIdentity.getId();
+    ExoSocialActivity activity = new ExoSocialActivityImpl();
+    activity.setTitle(activityTitle);
+    activity.setUserId(userId);
+    activityManager.saveActivityNoReturn(johnIdentity, activity);
+
+    long creatorId = Long.parseLong(johnIdentity.getId());
+    long audienceId = creatorId;
+
+    MetadataKey metadataKey = new MetadataKey(METADATA_TYPE_NAME, String.valueOf(FILE_ID), AUDIENCE_ID);
+    MetadataObject metadataObject = new MetadataObject(ExoSocialActivityImpl.DEFAULT_ACTIVITY_METADATA_OBJECT_TYPE,
+                                                       activity.getId(),
+                                                       PARENT_OBJECT_ID);
+    MetadataItem metadataItem = metadataService.createMetadataItem(metadataObject, metadataKey, creatorId);
+    assertNotNull(metadataItem);
+    assertTrue(metadataItem.getId() > 0);
+
+    List<Metadata> metadatas = metadataService.getMetadatas(METADATA_TYPE_NAME, -1);
+    assertNotNull(metadatas);
+    assertEquals(1, metadatas.size());
+
+    activityManager.deleteActivity(activity);
```

Comment:
Please add a test for file deletion

Reply:
no need because I removed ActivityAttachmentMetadataListener file!